core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -167,16 +148,46 @@ static ListeningScheduledExecutorService getDefaultExecutor() {
         MoreExecutors.getExitingScheduledExecutorService(executor));
   }
 
+  /** Extracts the Unix socket argument from specified properties object. If unset, returns null. */
+  private static String getUnixSocketArg(Properties props) {
+    String unixSocketPath = props.getProperty(UNIX_SOCKET_PROPERTY);
+    if (unixSocketPath != null) {
+      // Get the Unix socket file path from the properties object
+      return unixSocketPath;
+    } else if (System.getenv("CLOUD_SQL_FORCE_UNIX_SOCKET") != null) {
+      // If the deprecated env var is set, warn and use `/cloudsql/INSTANCE_CONNECTION_NAME`
+      // A socket factory is provided at this path for GAE, GCF, and Cloud Run
+      logger.warning(
+          String.format(
+              "\"CLOUD_SQL_FORCE_UNIX_SOCKET\" env var has been deprecated. Please use"
+                  + " '%s=\"/cloudsql/INSTANCE_CONNECTION_NAME\"' property in your JDBC url"
+                  + " instead.",
+              UNIX_SOCKET_PROPERTY));
+      return "/cloudsql/" + props.getProperty(CLOUD_SQL_INSTANCE_PROPERTY);
+    }
+    return null; // if unset, default to null
+  }
+
   /**
    * Creates a socket representing a connection to a Cloud SQL instance.
    *
-   * <p>Depending on the environment, it may return either a SSL Socket or a Unix Socket.
+   * @see CoreSocketFactory#connect(Properties, String)
+   */
+  public static Socket connect(Properties props) throws IOException {
+    return connect(props, null);
+  }
+
+  /**
+   * Creates a socket representing a connection to a Cloud SQL instance.
+   *
+   * <p>Depending on the given properties, it may return either a SSL Socket or a Unix Socket.
    *
    * @param props Properties used to configure the connection.
+   * @param unixPathSuffix Optional suffix to add the the Unix socket path. Unused if null.
    * @return the newly created Socket.
    * @throws IOException if error occurs during socket creation.
    */
-  public static Socket connect(Properties props) throws IOException {
+  public static Socket connect(Properties props, String unixPathSuffix) throws IOException {
     // Gather parameters
     final String csqlInstanceName = props.getProperty(CLOUD_SQL_INSTANCE_PROPERTY);
     final List<String> ipTypes = listIpTypes(props.getProperty("ipTypes", DEFAULT_IP_TYPES));
```

Comment:
You could move this down until after the unix socket block, since you don't use it until then.